pkgs/applications/misc/somebar/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, writeText
+, fetchurl
+, meson
+, ninja
+, pkg-config
+, wayland
+, pango
+, wayland-protocols
+, conf ? null
+, patches ? [ ]
+}:
+
+let
+  # There is a configuration in src/config.def.hpp, which we use by default
+  configFile = if lib.isDerivation conf || builtins.isPath conf then conf else "src/config.def.hpp";
+in
+
+stdenv.mkDerivation rec {
+  pname = "somebar";
+  version = "1.0.0";
+
+  src = fetchurl {
+    url = "https://git.sr.ht/~raphi/somebar/archive/${version}.tar.gz";
+    sha256 = "sha256-lCBjnwUeJ4R6jVy36KjFkUG7pn6WYA+Xkb0l43Fc0Mw=";
+  };
+
+  nativeBuildInputs = [ meson ninja pango pkg-config wayland wayland-protocols ];
+
+  # Allow for custom patches
+  inherit patches;
+
+  prePatch = "cp ${configFile} src/config.hpp";
+
+  meta = with lib; {
+    homepage = "https://git.sr.ht/~raphi/somebar";
+    description = "dwm-like bar for dwl";
+    license = licenses.mit;
+    maintainers = with maintainers; [ magnouvean ];
+    platforms = platforms.all;
```

Comment:
I don't think that's needed if you're not restricting the platforms.

Reply:
> I don't think that's needed if you're not restricting the platforms.

I guess you are referring to the platforms variable? I saw it done this way in another package so I thought this might have been standard, but you're probably correct. Thinking about it it doesn't really even make sense to package it for all platforms since `dwl` is only used on linux. Maybe I could set the `platforms` to `platforms.linux`?